### DISCUSSION TITLE
package version conflict error from cargo generate-lockfile command

### DISCUSSION THREAD

**Discussion by LinTechSo** [vectordotdev/vrl#547] [Q&A] [answered]
Hi,

Description:
When attempting to generate a lockfile using Cargo for Vector Dev Version 0.33.0 and VRL Version 0.8.1, the process encounters an issue related to resolving the lalrpop dependency version. The error message indicates that it failed to select a suitable version for the requirement lalrpop = "^0.20.1". The relevant details of the error are as follows:

```
cargo generate-lockfile
    Updating git repository `https://github.com/vectordotdev/vector`
    Updating crates.io index
    Updating git repository `https://github.com/vectordotdev/vrl`
error: failed to select a version for the requirement `lalrpop = "^0.20.1"`
candidate versions found which didn't match: 0.20.0, 0.19.12, 0.19.11, ...
location searched: crates.io index
required by package `vrl v0.8.0 (https://github.com/vectordotdev/vrl?rev=v0.8.0#fb02bfdd)`
    ... which satisfies git dependency `vrl` of package `vrl-server v0.1.0 (/Users/parham/work/hamravesh9/pipelineoperator-client/images/vrl_validator)`
perhaps a crate was updated and forgotten to be re-vendored?

```

Steps to Reproduce:

    Use Vector Dev Version 0.33.0.
    Use VRL Version 0.8.1.
    Run cargo generate-lockfile command.

Expected Behavior:
The cargo generate-lockfile command should complete successfully, resolving the lalrpop dependency to version ^0.20.1 without errors.

Actual Behavior:
The process encounters an error while attempting to select a version for the lalrpop requirement.

Additional Information:
my cargo.toml configuration:

```
[package]
name = "vrl-server"
version = "0.1.0"
edition = "2021"

[dependencies]
http = "0.2.9"
serde = { version = "1.0.188", default-features = false, features = ["derive"] }
serde_json = "1.0.107"
structopt = { version = "0.3.26", default-features = false }
tokio = { version = "1.32.0", default-features = false, features = ["full"] }
warp = "0.3.5"
pretty_env_logger = "0.4.0"

vrl = { git = "https://github.com/vectordotdev/vrl", rev = "v0.8.1" }
serde-wasm-bindgen = "0.5"
gloo-utils = { version = "0.1", features = ["serde"] }
getrandom = { version = "0.2", features = ["js"] }
vector-vrl-functions = { git = "https://github.com/vectordotdev/vector", tag = "v0.33.1"}
enrichment = { git = "https://github.com/vectordotdev/vector", tag = "v0.33.1"}
vector-common = { git = "https://github.com/vectordotdev/vector", tag = "v0.33.1"}
```

**Comment by jszwedko** [comment, marked as answer]
Hi @LinTechSo !

You'll need to make sure the Vector and VRL versions are compatible. In your case, v0.33.1 of Vector depends on v0.7.0 of VRL.
